### ISSUE TITLE
use feature does not get honored in quoted eval scope.

### ISSUE THREAD

**Issue by toddr** [rurban/perl-compiler#302]
The program:
```perl
use feature "lexical_subs";
no warnings 'experimental';

my $ret = eval q{my sub foo { "ok\n" }; foo()};
print $@ if($@);
print $ret;
```

as perl:
```sh
$>perl -e 'use feature "lexical_subs"; no warnings "experimental"; my $ret = eval q{my sub foo { "ok\n" }; foo()}; print $@ if($@); print $ret;'

ok
```

perlcc:
```sh
$>perlcc -e 'use feature "lexical_subs"; no warnings "experimental"; my $ret = eval q{my sub foo { "ok\n" }; foo()}; print $@ if($@); print $ret;'

Experimental "my" subs not enabled at (eval 1) line 1.
```

**Comment by toddr** [comment]
If lexical subs are throwing you off (they actually work right now) try this:

$>perlcc -r -e 'use feature "say"; eval q{say "ok"}; print $@ if($@);'
/usr/local/cpanel/3rdparty/perl/522-debug/bin/perlcc: Unexpected compiler output

String found where operator expected at (eval 1) line 1, near "say "ok""
	(Do you need to predeclare say?)
syntax error at (eval 1) line 1, near "say "ok""

**Comment by toddr** [comment]
Switching the quoted eval to an eval block does not have issues. Adding use feature "say" inside the quoted eval also fixes things.

**Comment by atoomic** [comment]
testc.sh 302

**Comment by toddr** [comment]
So it looks like %^H is used at compile time and discarded at run time? If so possibly the quoted eval is not getting it set via "hintseval?"

```
$>perl -MO=Concise  -e 'use feature "lexical_subs"; no warnings "experimental"; my $ret = eval q{my sub foo { "ok\n" }; foo()};  print $ret;'                                                                               
c  <@> leave[1 ref] vKP/REFC ->(end)
1     <0> enter ->2
2     <;> nextstate(main 77 -e:1) v:%,{,469762048 ->3
7     <2> sassign vKS/2 ->8
5        <1> entereval[t469893376] sK/HAS_HH,1 ->6
3           <$> const(PV "my sub foo { \"ok\\n\" }; foo()") s ->4
4           <$> hintseval(HV HASH) s ->5
6        <0> padsv[$ret:77,78] sRM*/LVINTRO ->7
8     <;> nextstate(main 78 -e:1) v:%,{,469762048 ->9
b     <@> print vK ->c
9        <0> pushmark s ->a
a        <0> padsv[$ret:77,78] s ->b
-e syntax OK
```

**Comment by rurban** [comment]
Fixed with 1.52_22 (hints_hash)
